src/main/java/com/example/onlinebookstore/service/order/OrderServiceImpl.java

```diff
@@ -0,0 +1,126 @@
+import com.example.onlinebookstore.mapper.OrderItemMapper;
+import com.example.onlinebookstore.mapper.OrderMapper;
+import com.example.onlinebookstore.model.Book;
+import com.example.onlinebookstore.model.CartItem;
+import com.example.onlinebookstore.model.Order;
+import com.example.onlinebookstore.model.OrderItem;
+import com.example.onlinebookstore.model.ShoppingCart;
+import com.example.onlinebookstore.model.User;
+import com.example.onlinebookstore.repository.CartItemRepository;
+import com.example.onlinebookstore.repository.OrderItemRepository;
+import com.example.onlinebookstore.repository.OrderRepository;
+import com.example.onlinebookstore.repository.ShoppingCartRepository;
+import com.example.onlinebookstore.repository.UserRepository;
+import jakarta.transaction.Transactional;
+import java.math.BigDecimal;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+    private final OrderRepository orderRepository;
+    private final OrderMapper orderMapper;
+    private final OrderItemRepository orderItemRepository;
+    private final OrderItemMapper orderItemMapper;
+    private final UserRepository userRepository;
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final CartItemRepository cartItemRepository;
+
+    @Override
+    public List<OrderResponseDto> getAllOrders(Long userId) {
+        List<Order> orders = orderRepository.findAllByUserId(userId);
+        return orders.stream().map(orderMapper::toDto).toList();
+    }
+
+    @Override
+    public List<OrderItemResponseDto> getOrderItemsByOrderId(Long orderId) {
+        List<OrderItem> orderItems = orderItemRepository.findAllByOrderId(orderId);
+        return orderItems.stream().map(orderItemMapper::toDto).toList();
+    }
+
+    @Override
+    public OrderItemResponseDto getOrderItem(Long orderId, Long itemId) {
+        OrderItem orderItem = orderItemRepository.findByOrderIdAndId(orderId, itemId)
+                .orElseThrow(() -> new EntityNotFoundException(
+                        "Cant find order item with id " + itemId + " and order id " + orderId));
+        return orderItemMapper.toDto(orderItem);
+    }
+
+    @Override
+    @Transactional
+    public OrderResponseDto updateStatus(Long id, PatchOrderRequestDto requestDto) {
+        Order order = orderRepository.findById(id)
+                .orElseThrow(() -> new EntityNotFoundException(
+                        "Can't find order with id " + id));
+        order.setStatus(Order.Status.valueOf(requestDto.getStatus()));
+        orderRepository.save(order);
+        return orderMapper.toDto(order);
+    }
+
+    @Override
+    @Transactional
+    public OrderResponseDto placeOrder(Long userId, PostOrderRequestDto requestDto) {
+        User user = userRepository.findById(userId)
```

Comment:
I'm not sure that you need to fetch a user from DB as you are fetching a shopping cart by userId anyway.